extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java

```diff
@@ -172,7 +178,12 @@ public void buildReactivePersistenceUnit(
                         .orElse(!dataSourcesBuildTimeConfig.hasNamedDataSources()),
                 curateOutcomeBuildItem);
 
-        if (dbKindOptional.isPresent()) {
+        if (dbKindOptional.isEmpty()) {
```

Comment:
Is it true that it needs to be configured? Isn't that only true in PROD mode?
In DEV mode you get DEV services.

The real problem that should halt the build is when there's no DB driver, as mentioned by @yrodiere, I think.

Reply:
I have no idea TBH. We could only throw if in PROD mode... 

> The real problem that should halt the build is when there's no DB driver, as mentioned by @yrodiere, I think.

This is definitely not my area of expertise so if anyone tells me how to do it I'd be more than happy to follow.

Reply:
Ok, so it should be fine as the `dbKindOptional` is actually resolve a few lines above - https://github.com/quarkusio/quarkus/blob/main/extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java#L168-L173. In other words, the optional is not empty for example if in the DEV mode and dev service is used.

Reply:
This implementation is quite an approximation, and it may not work in some edge cases, though I couldn't say which ones without spending more time on this.

I'd recommend having reactive clients publish a build item similar to `JdbcDataSourceBuildItem` and using that to determine whether a datasource is configured or not, exactly like we do for Hibernate ORM.

Reply:
Hopefully we'll refine this in https://github.com/quarkusio/quarkus/issues/39049